iroh-store/src/store.rs

```diff
@@ -367,9 +367,9 @@ impl<'a> WriteStore<'a> {
         let mut total_blob_size = 0;
 
         let mut batch = WriteBatch::default();
-        let mut cid_tracker: HashSet<Cid> = std::collections::HashSet::default();
+        let mut cid_tracker: AHashSet<Cid> = AHashSet::default();
```

Comment:
If we had a vec we could use https://docs.rs/ahash/latest/ahash/struct.AHashSet.html#method.with_capacity here to make sure the hashset does not need to rehash.

But with the iterator we could just use the size hint, which is probably not worth it.

Reply:
`HashSet`s have `with_capacity`: https://docs.rs/ahash/latest/ahash/struct.AHashSet.html#method.with_capacity